### ISSUE TITLE
inspeqtorctl fails with nc: invalid option -- 'U' (workaround possible)

### ISSUE THREAD

**Issue by tibra** [mperham/inspeqtor#163]
Dear Mike 👋,

calling the inspeqtorctl command fails:
```
inspeqtorctl
> nc: invalid option -- 'U'
```

**Environment**
```
Debian 10 Buster
Inspeqtor 2.0.0
netcat v1.10-41.1
```

[Other](https://github.com/ymnk/jsch-agent-proxy/issues/31) [projects](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=614291) seem to have similar problems.

It seems to be related to line 13 [here](https://github.com/mperham/inspeqtor/blob/995453a24bc7b987b379f99600c5c02b7ccc8b40/packaging/root/usr/bin/inspeqtorctl):
```
if [ ! -x /bin/nc ] && [ ! -x /usr/bin/nc ]; then
  # Some OS's ship with nc in the base, others don't.
  # What a crock.
  echo Could not find a working netcat in /bin/nc or /usr/bin/nc.
  echo Please install your operating system\'s \'netcat\' or \'nc\' package
else
  echo "$@" | nc -U ${INSPEQTOR_SOCK:-/var/run/inspeqtor.sock}
fi
```

**🔧 Workaround**
I can confirm, [as suggested here](https://github.com/ymnk/jsch-agent-proxy/issues/31#issuecomment-466620935), that the `netcat-openbsd` package on debian works, which supports the old cli and inspeqtorctl works as intended.

I can live with the workaround, but others may be wondering what goes on. Maybe you have an idea how to fix it without having another conditional.

Thank you kindly!

**Comment by mperham** [comment]
If I remove -U, will `nc` start failing on other platforms that do support `-U`?

**Comment by tibra** [comment]
Yes, it will then fail without providing a conditional or another solution to read the socket because the CLI changed (an since nowadays container images change their base images too, it might create regressions):

Options, that nc.traditional does not know are: `-4 / -6, -U, -k`
Options, that nc6 does not know are: `-U, -k` but it knows `--continuous`

- Creating a conditional on inspeqtor's side is not straight forward and probably error prone because checking the version number r[equires parsing the man pages](https://unix.stackexchange.com/questions/537016/how-can-i-check-what-version-of-netcat-am-i-running) m(
- Requiring `netcat-openbsd` as a dependency would be an option, albeit contrary to the philosophy of a standalone library.
- Apart from that, [socat](http://www.dest-unreach.org/socat/) seems to be an actively maintained and modern alternative.

Would it make sense to bundle this into the inspeqtor package if the OS cli env is not reliable enough?

**Comment by mperham** [comment]
Yeah we can change the packages and binary to install and use socat.

**Comment by harmonydzen-blip** [comment]
Hi @mperham !

Can you write example how you change socat Instead of nc in service files ? 
[cloud-init-local.service](https://github.com/canonical/cloud-init/blob/25.1.x/systemd/cloud-init-local.service.tmpl#L35)
```

ExecStart=sh -c 'echo "start" | nc -Uu -W1 /run/cloud-init/share/local.sock -s /run/cloud-init/share/local-return.sock | sh'

```
For example in this service ?
